cmd/proxy/wsl_integration_linux.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright © 2024 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	debug        bool
+	upstreamAddr string
+	listenAddr   string
+)
+
+const (
+	k8sAPI            = "http://192.168.1.2:6443"
+	defaultListenAddr = "127.0.0.1:6443"
+)
+
+func main() {
+	flag.BoolVar(&debug, "debug", false, "enable additional debugging")
+	flag.StringVar(&upstreamAddr, "upstream-addr", k8sAPI, "The upstream server's address.")
+	flag.StringVar(&listenAddr, "listen-addr", defaultListenAddr, "The server's address in an IP:PORT format.")
+	flag.Parse()
+
+	if debug {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+
+	targetURL, err := url.Parse(upstreamAddr)
+	if err != nil {
+		logrus.Fatalf("invalid upstream URL: %s", upstreamAddr)
+	}
+
+	proxy := requestHandler{
+		TargetURL: targetURL,
+	}
+
+	http.HandleFunc("/", proxy.handleRequestAndRedirect)
```

Comment:
It's unclear why this is acting on layer 7 instead of layer 4 (that is, why is this not just blindly slinging TCP connections).

Reply:
@Nino-K said that we don't really want to deal with making this generic for now; I guess that's okay, but we may need to expand it in the future (which is more work because we'd need to have a system to figure out which ports should be forwarded).